snackpub/src/index.ts

```diff
@@ -40,23 +47,69 @@ function closeServerAsync(server: Server): Promise<void> {
   });
 }
 
+function terminateSocket(
+  socket: Socket<ClientToServerEvents, ServerToClientEvents, InterServerEvents, SocketData>,
+  reason: string
+) {
+  socket.emit('terminate', reason);
+
+  // When clients receive the `terminate` message, they should disconnect and no longer try to reconnect back to the server.
+  // In case the clients do not handle the `terminate` message, e.g. from other websocket clients other than snackpub clients.
+  // We will wait three seconds to close the connections actively.
+  setTimeout(() => {
+    if (socket.connected) {
+      socket.disconnect(true);
+    }
+  }, 3000);
```

Comment:
While it's not necessary, is there a way to listen to disconnection events from the socket and call `clearTimeout` if the client disconnects? That way we don't have unnecessary setTimeout handles.